### PR TITLE
Add capability to return snapshots in SQLite format

### PR DESCRIPTION
**Pull request by gbrail**
Unfortunately I had to do a "vendor" update here so there are extra commits. But if you focus on "sqlite.go" and its dependencies then you can see how the snapshots will work.

### CODE REVIEW

File: snapshotserver/pgschema.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
+package main
+
+import "database/sql"
+
+// A pgColumn describes a single column in a table
+type pgColumn struct {
+	name  string
+	typid int
```
Comment (alexkhimich) [inline snapshotserver/pgschema.go:23]: typeid ?

File: snapshotserver/schema_test.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"fmt"
+)
+
+var _ = Describe("Schema tests", func() {
```
Comment (alexkhimich) [inline snapshotserver/schema_test.go:26]: Could not find the test data setup step. Where is it?

File: snapshotserver/snapshot_info_test.go [hunk 3]
```diff
@@ -47,7 +47,7 @@ var (
 func TestSnapshot(t *testing.T) {
 	dbURL = os.Getenv("TEST_PG_URL")
 	if dbURL == "" {
```
Comment (alexkhimich) [inline snapshotserver/snapshot_info_test.go:53]: Bad idea to skip tests in such unpredictable way.

File: snapshotserver/types.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	case sqlInteger:
+		return "integer"
+	case sqlReal:
+		return "real"
+	case sqlText:
+		return "text"
+	case sqlBlob:
+		return "blob"
+	default:
+		panic(fmt.Sprintf("Invalid type %d", t))
```
Comment (alexkhimich) [inline snapshotserver/types.go:45]: Wonder why here we use sprintf but elsewhere debug logs

File: glide.lock [hunk 5]
```diff
@@ -1,16 +1,16 @@
 hash: bcddc87691b59c9b684ca222ba35984a22fa98b07d38f2e68d06381695ed4002
```
Comment (alexkhimich) [inline glide.lock:1]: Maybe .gitignore it?

File: snapshotserver/snapshot_api_test.go [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+	"path"
+
+	"time"
+
+	"github.com/apigee-labs/transicator/common"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Snapshot API Tests", func() {
```
Comment (alexkhimich) [inline snapshotserver/snapshot_api_test.go:34]: Missing negative tests.

File: snapshotserver/schema_test.go [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Schema tests", func() {
```
Comment (alexkhimich) [inline snapshotserver/schema_test.go:26]: Negative tests please

File: snapshotserver/sqlite.go [hunk 8]
```diff
@@ -0,0 +1,292 @@
...
+			if err != nil {
+				sendAPIError(serverError, err.Error(), w, r)
+				return err
+			}
+		} else {
+			log.Debugf("Skipping table %s which has no selector", tid)
+		}
+	}
+
+	pgTx.Commit()
```
Comment (alexkhimich) [inline snapshotserver/sqlite.go:96]: Do you need this if you defer it at line #72

File: snapshotserver/sqlite.go [hunk 9]
```diff
@@ -0,0 +1,292 @@
...
+
+	// Close the database. Checkpoint the WAL so that the result
+	// is a single file, with no additional .wal or .shm file.
+	_, err = tdb.Exec("pragma wal_checkpoint(TRUNCATE)")
+	if err != nil {
+		sendAPIError(serverError, err.Error(), w, r)
+		return err
+	}
+	tdb.Close()
```
Comment (alexkhimich) [inline snapshotserver/sqlite.go:105]: same here, there is defer of this

**Review by alexkhimich — COMMENTED** [review]
Pls add "vendor" and other glide things to .gitignore.

**Comment by gbrail** [comment]
Good comments -- will revise. Thanks!

**Comment by alexkhimich** [comment]
Have u just merged code in master without addressing review feedback? 
Maybe there is other branch somewhere?

**Pull request closed without merging**